src/shared/calendarspec.c

```diff
@@ -872,6 +913,38 @@ static int parse_calendar_time(const char **p, CalendarSpec *c) {
         return 0;
 }
 
+static int parse_and_every(const char *s, usec_t *ts, usec_t *rep) {
+        _cleanup_free_ char *p_ts = NULL;
+        int i, r;
+
+        char const *p = strcasestr(s, " and ");
+        if ( !p )
+                return -EINVAL;
+
+        p_ts = strndup(s, p - s);
+        i = strlen(p_ts);
+        while ( i > 0 && isspace(p_ts[--i]) )
+                p_ts[i] = (char)0;
+
+        r = parse_timestamp(p_ts, ts);
```

Comment:
Please store a value to result buffer only when the function succeed.
Hence, please introduce a temporary variable and assign to ret_ts on success at the end of the function.